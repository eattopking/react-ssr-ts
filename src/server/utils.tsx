import React from "react";
import PageLayout from "../containers/Home/index";
import Login from "../containers/Login/index";
import Register from "../containers/Register/index";
import { Provider } from "react-redux";
import { renderToString } from "react-dom/server";
import { StaticRouter } from "react-router-dom";

/**
 * 用于产生主页服务端渲染字符串的方法
 */
export function render({ url, context = {}, store }: { url: string; context: object; store: any }) {
  /**
   * location={url}作用应该是每次后台初次返回页面时决定,初次显示那个路径下的页面
   * 使用window.context将数据缓存起来,实现数据注水,供同构的client的js代码引用
   */
  const content = renderToString(
    <Provider store={store}>
      <StaticRouter location={url} context={context}>
        <PageLayout />
      </StaticRouter>
    </Provider>
  );
  return `<!DOCTYPE html>
  <html lang="zh-CN">
    <head>
      <title>ssrTable</title>
      <link rel="stylesheet" href="/public/index.css">
    </head>
    <body>
      <div id="app">${content}</div>
      <script>
      window.context = {
        state: ${JSON.stringify(store.getState())}
      }
      </script>
      <script src="/public/index.js"></script>
    </body>
  </html>`;
}

/**
 * 用于产生登录页面服务端渲染字符串的方法
 */
export function loginRender() {
  /**
   * 使用window.context将数据缓存起来,实现数据注水,供同构的client的js代码引用
   */
  const content = renderToString(<Login />);
  return `<!DOCTYPE html>
  <html lang="zh-CN">
    <head>
      <title>登录界面</title>
      <link rel="stylesheet" href="login.css">
    </head>
    <body>
      <div id="app">${content}</div>
      <script src="/public/login.js"></script>
    </body>
  </html>`;
}

/**
 * 用于产生注册页面服务端渲染字符串的方法
 */
export function registerRender() {
  /**
   * 使用window.context将数据缓存起来,实现数据注水,供同构的client的js代码引用
   */
  const content = renderToString(<Register />);
  return `<!DOCTYPE html>
  <html lang="zh-CN">
    <head>
      <title>注册界面</title>
      <link rel="stylesheet" href="register.css">
    </head>
    <body>
      <div id="app">${content}</div>
      <script src="/public/register.js"></script>
    </body>
  </html>`;
}
