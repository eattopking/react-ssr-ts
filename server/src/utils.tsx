import React from "react";
import PageLayout from "./containers/Home/index";
import { Provider } from "react-redux";
import { renderToString } from "react-dom/server";
import { StaticRouter } from "react-router-dom";

// 用于产生服务端渲染字符串的方法
export default function render({ url, context = {}, store }: { url: string; context: object; store: any }) {
  /**
   * location={url}作用应该是每次后台初次返回页面时决定,初次显示那个路径下的页面
   * 使用window.context将数据缓存起来,实现数据注水,供同构的client的js代码引用
   */
  const content = renderToString(
    <Provider store={store}>
      <StaticRouter location={url} context={context}>
        <PageLayout />
      </StaticRouter>
    </Provider>
  );
  return `<!DOCTYPE html>
  <html lang="zh-CN">
    <head>
      <title>ssrTable</title>
      <link rel="stylesheet" href="/index.css">
    </head>
    <body>
      <div id="app">${content}</div>
      <script>
      window.context = {
        state: ${JSON.stringify(store.getState())}
      }
      </script>
      <script src="/index.js"></script>
    </body>
  </html>`;
}
