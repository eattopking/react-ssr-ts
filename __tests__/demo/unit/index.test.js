import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../../src/demo/Header/index';
import { getElement } from '../../../src/utils/testUtils';

// 测试代码  单元测试的形式
describe('单元测试header 组件', () => {
    console.log(100000000)
    it('内容是不是123', () => {
        // 获取浅层Header组件，就是只能获取Header内部， Header的子元素内部获取不到， 需要获取需要用mount
        const wrapper = shallow(<Header />);
        const element = getElement(wrapper, 'div');
        expect(element.text()).toBe('023');
    });

    it('onClick 事件掉用之后，state 的name属性是不是666', () => {
        const wrapper = shallow(<Header />);

        const element = getElement(wrapper, 'div');
        const element1 = getElement(wrapper, 'div1');
        // 主动触发元素上注册的事件
        element.simulate('click', {
            key: 13
        });
        expect(element1.text()).toBe('0');
    });

    it('内容没有变化测试', () => {
        const wrapper = shallow(<Header />);
        // 使用快照测试我们组件的整体结构有没有变化
        expect(wrapper).toMatchSnapshot();
    })
})



// wrapper.instance().handelClick是获取类组件内部的this， 从而获取类组件中的方法
// wrapper.prop('children')是获取类组件内部的props，
// wrapper.state('name')是获取类组件内部的state，
// wrapper.setState()调用类组件的setState
// 获取的组件索引要通过List.at(索引)， 获取对应的元素
// 获取的组件索引要通过wrapper.find('Footer')， 获取组件内部的元素和组件
// 更新组件最新内容wrapper.update()
// 获取组件dom结构wrapper.debug()
