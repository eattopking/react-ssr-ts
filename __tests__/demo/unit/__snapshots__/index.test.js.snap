// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`单元测试header 组件 内容没有变化测试 1`] = `
<div
  className="test"
  data-test="div"
  onClick={[Function]}
>
  <div
    data-test="div1"
    key="1"
  >
    0
  </div>
  <div
    key="2"
  >
    2
  </div>
  <div
    key="3"
  >
    3
  </div>
</div>
`;
